File: Leet-Code/412_Fizz_Buzz.cpp
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        
        vector <string> answer;
        for(int i = 1; i <= n; i++) {
            string s = "";
            if (i % 3 == 0) 
                s += "Fizz";
            
            if (i % 5 == 0) 
                s += "Buzz";
        
            if (i % 3 != 0 && i % 5 != 0)
                s += to_string(i);
            
            answer.push_back(s);
        }
        
        return answer;

        // vector <string> answer;
        // for(int i = 1; i <= n; i++) {
        //     if ((i % 15) == 0) 
        //         answer.push_back("FizzBuzz");
        //     else if ((i % 3) == 0)
        //         answer.push_back("Fizz");
        //     else if ((i % 5) == 0)
        //         answer.push_back("Buzz");
        //     else answer.push_back(to_string(i));
        // }
        
        // return answer;

    }
};

File: Leet-Code/412_Fizz_Buzz.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        
        ans = []
        for i in range(n):
            s = ''
            if ((i + 1) % 3 == 0):
                s += 'Fizz'

            if ((i + 1) % 5 == 0):
                s += 'Buzz'
            
            if ((i + 1) % 3 != 0 and (i + 1) % 5 != 0):
                s += str(i+1)
            
            ans.append(s)

        return ans


        # l=[]
        # for i in range(1,n+1):
        #     a=(i%3==0)
        #     b=(i%5==0)
        #     l.append(str(i)*(not(a or b))+"Fizz"*(a)+"Buzz"*b)
        # return l


        # liste=[str(i) for i in range(1,n+1)]
        # k=1  
        # while 3*k-1 < n : #check condition
        #     liste[3*k-1]="Fizz"
        #     k+=1
        # l=1
        # while 5*l-1 < n:
        #     liste[5*l-1]="Buzz"
        #     l+=1
        # m=1
        # while 15*m-1<n:
        #     liste[15*m-1]="FizzBuzz"
        #     m+=1
        # return liste